Detectors/TOF/base/src/Geo.cxx

```diff
@@ -19,3 +20,478 @@ constexpr Float_t Geo::HEIGHTS[NPLATES][NMAXNSTRIP];
 constexpr Float_t Geo::DISTANCES[NPLATES][NMAXNSTRIP];
 constexpr Bool_t Geo::FEAWITHMASKS[18];
 constexpr Float_t Geo::ROOF2PARAMETERS[3];
+
+void Geo::GetDetID(Float_t* pos, Int_t* det)
+{
```

Comment:
Hi Ruben,
this point is a bit tricky because the GetDetID method maps from position real Float[3] to Integer[5].
While for the inverse mapping we have a discrete set of values in this method it is not possible to cache the indexes beacuse the inputs (Float[3]) are Always different.
I manage the code in order to avoid multiple calls of the same routines (see below)